### PR TITLE
try out scalike for job cost dao

### PR DESCRIPTION
**Pull request by dvoet**
an example of how we might use scalikeJDBC. Compare insertCallCostSql2 and getCallCostSql2 in JobCostDAO with the doobie versions. Note we don't specify column names as they are automatically converted from the camel case convention in scala to the snake case convention in databases (although we could define it if we wanted).

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/workbench/hamm/db/JobCostDAO.scala [hunk 1]
```diff
@@ -79,10 +98,57 @@ object JobCostDAO {
...
-final case class JobCost(workflowId: WorkflowId, callFqn: CallFqn, attempt: Short, jobIndexId: Int, gcpJobId: Option[String], startTime: Instant, endTime: Instant, cost: Double){
-  val uniqueKey = CallUniquekey(workflowId, callFqn, attempt, jobIndexId)
+final case class JobCost(workflowId: WorkflowId, callFqn: CallFqn, attempt: Short, jobIndex: Int, vendorJobId: Option[String], startTime: Instant, endTime: Instant, cost: Double){
+  val uniqueKey = CallUniquekey(workflowId, callFqn, attempt, jobIndex)
+}
+
+object JobCost extends SQLSyntaxSupport[JobCost] {
+  override def tableName: String = "JOB_COST"
+  import Binders._
+  def apply(e: ResultName[JobCost])(rs: WrappedResultSet): JobCost = JobCost(
```
Comment (dvoet) [inline core/src/main/scala/org/broadinstitute/workbench/hamm/db/JobCostDAO.scala]: method for converting a result set to a JobCost object

File: core/src/main/scala/org/broadinstitute/workbench/hamm/db/JobCostDAO.scala [hunk 2]
```diff
@@ -79,10 +98,57 @@ object JobCostDAO {
...
+    rs.get(e.attempt),
+    rs.get(e.jobIndex),
+    rs.get(e.vendorJobId),
+    rs.get(e.startTime),
+    rs.get(e.endTime),
+    rs.get(e.cost)
+  )
 }
+
+object Binders {
```
Comment (dvoet) [inline core/src/main/scala/org/broadinstitute/workbench/hamm/db/JobCostDAO.scala]: methods for binding custom types to parameters and from results

File: core/src/test/scala/org/broadinstitute/workbench/hamm/db/JobCostDAOTest.scala [hunk 3]
```diff
@@ -3,24 +3,48 @@ package org.broadinstitute.workbench.hamm.db
...
+//          } yield {
+//            assertEquals(retrievedObject, jobCost)
+//          }
+//
+//          res.unsafeRunSync()
+//          true
+//      }
+//    }
+//
+  test("sqllike") {
```
Comment (dvoet) [inline core/src/test/scala/org/broadinstitute/workbench/hamm/db/JobCostDAOTest.scala]: example usage

File: core/src/main/scala/org/broadinstitute/workbench/hamm/db/JobCostDAO.scala [hunk 4]
```diff
@@ -79,10 +98,57 @@ object JobCostDAO {
...
+
+  def getCallCostSql2(callUniquekey: CallUniquekey)(implicit session: DBSession) = {
+    import Binders._
+    val e = JobCost.syntax("e")
+    withSQL {
+      select.from(JobCost as e)
+        .where.eq(e.workflowId, callUniquekey.workflowId)
+        .and.eq(e.callFqn, callUniquekey.callFqn)
+        .and.eq(e.attempt, callUniquekey.attempt)
+    }.map(JobCost(e.resultName)).single().apply()
```
Comment (ansingh7115) [inline core/src/main/scala/org/broadinstitute/workbench/hamm/db/JobCostDAO.scala]: I like not using raw sql  with appended strings.

File: core/src/main/scala/org/broadinstitute/workbench/hamm/db/JobCostDAO.scala (hunk 1 above)
Comment (Qi77Qi) [reply core/src/main/scala/org/broadinstitute/workbench/hamm/db/JobCostDAO.scala]: what happens if `e` doesn't have one of the field?

**Comment by dvoet** [comment]
Compile error

**Pull request closed without merging**